linera-core/src/client/mod.rs

```diff
@@ -1736,25 +1736,22 @@ where
         round: Round,
         manager: ChainManagerInfo,
     ) -> Result<Certificate, ChainClientError> {
-        let next_block_height;
-        let block_hash;
-        {
+        let (next_block_height, block_hash, tip_time) = {
             let state = self.state();
-            next_block_height = state.next_block_height();
-            block_hash = state.block_hash();
             // In the fast round, we must never make any conflicting proposals.
             if round.is_fast() {
                 if let Some(pending) = &state.pending_block() {
                     ensure!(
                         pending == &block,
                         ChainClientError::BlockProposalError(
                             "Client state has a different pending block; \
-                             use the `linera retry-pending-block` command to commit that first"
+                            use the `linera retry-pending-block` command to commit that first"
                         )
                     );
                 }
             }
-        }
+            state.tip()
```

Comment:
Definitely create a data structure so that we can write `tip.timestamp` instead of `tip_time` etc